File: src/167TwoSumIIInputArrayIsSorted/167TwoSumIIInputArrayIsSorted.go
package main

func twoSum(numbers []int, target int) []int {
	n := len(numbers)
	lo := 0
	hi := n - 1
	for lo < hi {
		sum := numbers[lo] + numbers[hi]
		if sum == target {
			return []int{lo + 1, hi + 1}
		} else if sum < target {
			lo++
		} else {
			hi--
		}
	}
	return nil
}

func main() {

}

File: src/167TwoSumIIInputArrayIsSorted/167TwoSumIIInputArrayIsSorted.rs
impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let n = numbers.len() as i32;
        let mut i = 0;
        let mut j = n-1;
        while i < j {
            let sum = numbers[i as usize]+numbers[j as usize]; 
            if sum == target {
                return vec![i+1, j+1];
            } else if sum > target {
                j -= 1;
            } else {
                i += 1;
            }
        }
        vec![]
    }
}
